### PR TITLE
Feat: add preview in export dialog + fix clipboard

### LINKED ISSUE #1005: Feature: Add JSON preview and copy to clipboard in Export as .cv File dialog
### Is your feature request related to a problem? Please describe

The **Export as .cv File** dialog (`v0/src/components/helpers/ExportProject.vue`) only offers a filename input and a Save/Download button. Users have no way to inspect the serialised circuit JSON before saving — they must download the file and open it in a separate text editor to see its contents. This is especially 
painful when debugging serialisation issues or sharing a circuit snippet quickly.

Additionally, the shared `copyToClipboard` utility (`v0/src/simulator/src/utils.ts`) silently fails inside Vuetify focus-trapped dialogs because `document.execCommand('copy')` requires direct document focus that focus-trapped dialogs break. There is no fallback or error surfaced to the user.

### Describe the solution you'd like

Enhance the Export as .cv File dialog with:

- A **Preview** button that calls `generateSaveData()` and renders the circuit   JSON in a read-only CodeMirror editor (JSON syntax highlighting + line numbers)
- A **Refresh Preview** button to regenerate the preview after circuit edits
- A **Copy to Clipboard** button that copies the JSON directly to the system clipboard without requiring a file save 
- Proper **loading**, **error**, and **empty-state** handling throughout

Additionally, update `copyToClipboard` in `utils.ts` to use the modern `navigator.clipboard.writeText` API as the primary path, with `execCommand` retained only as a legacy fallback — fixing the silent failure inside dialogs.

### Describe alternatives you've considered

_No response_

### Is requested feature related to vue simulator?

Yes — the Export as .cv File dialog and the copyToClipboard utility are both part of the Vue simulator. 

### Additional context

_No response_

### Are you working on this?

Yes

### PR DESCRIPTION
**Pull request by Harkeerat24**
Fixes #1005

#### Describe the changes you have made in this PR -

**Commit 1 — Feature: JSON Preview in Export Dialog** (`src/components/DialogBox/ExportProject.vue`)
- Added a **Preview** button using the same `Codemirror` editor component as `ExportVerilog` (same dimensions, border, height/width pattern)
- Circuit data is now stringified via `JSON.stringify()` before passing to `downloadFile`/`copyToClipboard` — previously the raw JS object was passed, resulting in `[object Object]` in saved files and clipboard
- Added loading spinner (`v-progress-circular`), error state, and empty-data guard (`v-else-if="previewData"`) for the preview panel
- Cached preview data is reused for Save/Copy when already generated (avoids redundant `generateSaveData` calls)
- Default filename simplified to project name only (matches original repo's SaveAs.js behavior)
- Button order: **Preview → Save → Copy to Clipboard**

**Commit 2 — Fix: Clipboard utility** (`src/simulator/src/utils.ts`)
- `copyToClipboard` now uses `navigator.clipboard.writeText()` as the primary path, with legacy `document.execCommand("copy")` as fallback
- Made the function `async` to support `await` on the modern Clipboard API
- Discovered while testing: Vuetify's `v-dialog` focus-trap can cause `execCommand` to silently fail in the standalone Vite dev server. The modern API is unaffected. This does not impact production (CircuitVerse.org).

### Screenshots of the UI changes (If any) -

https://github.com/user-attachments/assets/74bbd1f7-824a-4088-88c1-8c771e03ecdf

---

## Code Understanding and AI Usage

**Did you use AI assistance (ChatGPT, Claude, Copilot, etc.) to write any part of this code?**
- [x] No, I wrote all the code myself
- [ ] Yes, I used AI assistance (continue below)

**Explain your implementation approach:**

**Approach:** I followed the existing `ExportVerilog.vue` pattern — same `Codemirror` component (`codemirror-editor-vue3`), same `height: 300` / `width: 700`, same `messageBtn block` button layout. This keeps the UI consistent across export dialogs.

## Checklist before requesting a review
- [x] I have added proper PR title and linked to the issue
- [x] I have performed a self-review of my code
- [x] **I can explain the purpose of every function, class, and logic block I added**
- [x] I understand why my changes work and have tested them thoroughly
- [x] I have considered potential edge cases and how my code handles them
- [ ] If it is a core feature, I have added thorough tests
- [x] My code follows the project's style guidelines and conventions

## Summary by CodeRabbit

* **New Features**
  * Added a Preview section with syntax-highlighted code display in the export dialog, plus loading and error states.
  * Added Preview/Refresh, Save, and Copy to Clipboard action buttons; export filename now defaults to the project name.
* **Improved Reliability**
  * Clipboard copy now uses a more robust, cross-browser approach for better success rates.

### CODE REVIEW

File: src/components/DialogBox/ExportProject.vue [hunk 1]
```diff
@@ -57,31 +116,122 @@ export function ExportProject() {
...
+    isLoading.value = false
+    SimulatorState.dialogBox.export_project_dialog = false
+}
+
+async function generateCircuitDataString(): Promise<string | null> {
+    try {
+        const data = await generateSaveData(
+            projectStore.getProjectName,
+            false
+        )
```
[the remark this answers is not in the document]
Comment (Harkeerat24) [reply src/components/DialogBox/ExportProject.vue:160]: generateSaveData runs update routines (updateSubcircuitSet, update) to ensure subcircuit integrity before serializing. This is intentional — the same function is used by Save/Copy and is the only correct serialization path in the codebase.

**Comment by Harkeerat24** [comment]
@Nihal4777 @ThatDeparted2061 @tachyons please review :)

**Comment by tachyons** [comment]
Issue is still pending triage
